### PR TITLE
Tank wrecks

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Tanks when destroyed now leave a wreck behind.

image

The wrecks behave like the tank object otherwise, and can itself be destroyed entirely.

APC doesn't have a wreck because I didn't want to make a wrecked sprite for the god ugly m113 apc we have (Tye please resprite it).

There is a proc to repair wrecked tanks, but it won't be available outside of adminnery for the time being.

~~For mappers, these can be spawned directly to use them as map props if desired.~~
Mappers, now that this PR is actually wrecking the tank object, I will do some mapping helper or something to allow this for mapping purposes.

## Why It's Good For The Game
Looks cool, and is better than the tank just vanishing in a puff of smoke.
## Changelog
:cl:
add: Tanks leave behind wrecks when destroyed
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
Will mobs inside the tank just spawn inside the structure on destruction? It could lead to someone hiding in it to gain advantage I think.

**Comment by Tatarla** [comment]
Rutgmc aka Unga Marines expresses gratitude for the work done We will port this miracle

**Comment by Woker2004** [comment]
I think that this should also add a way to repair tank to functionality via req shit and engineers, because AC's on tank death just go immediately suicide into xenos when it's gone

**Review by Kuro020 — APPROVED** [review]

**Comment by Lumipharon** [comment]
Setting to draft, almost finished the new implementation.

File: code/modules/vehicles/armored/__armored.dm [hunk 1]
```diff
@@ -671,8 +672,7 @@
 	var/image/secondary_overlay
 
 /atom/movable/vis_obj/turret_overlay/Destroy()
-	if(primary_overlay)
-		QDEL_NULL(primary_overlay)
+	QDEL_NULL(primary_overlay)
```
Comment (TiviPlus) [inline code/modules/vehicles/armored/__armored.dm:675]: Doesn't this mean runtimes due to null passes to qdel is possible

File: code/modules/vehicles/armored/_multitile.dm [hunk 2]
```diff
@@ -28,6 +28,40 @@
...
+	name = initial(name)
+	if(armored_flags & ARMORED_IS_WRECK)
+		name = "wrecked " + name
+
+/obj/vehicle/sealed/armored/multitile/update_desc(updates)
+	. = ..()
+	desc = initial(desc)
+	if(armored_flags & ARMORED_IS_WRECK)
+		desc += " Now just a smouldering ruin."
+
```
Comment (TiviPlus) [inline code/modules/vehicles/armored/_multitile.dm:51]: Correct me if I'm wrong but shouldn't this be editing the return

Comment (TiviPlus) [reply code/modules/vehicles/armored/_multitile.dm:51]: Name too

**Review by TiviPlus — COMMENTED** [review]
Rest looks ok

File: code/modules/vehicles/armored/_multitile.dm (hunk 2 above)
Comment (Lumipharon) [reply code/modules/vehicles/armored/_multitile.dm:51]: Nah, these actually just direct set it. Parent call is just for the sig

File: code/modules/vehicles/armored/__armored.dm (hunk 1 above)
Comment (Lumipharon) [reply code/modules/vehicles/armored/__armored.dm:675]: nulls don't cause runtimes in qdel

**Pull request merged**